compiler/optimizer/VPHandlers.cpp

```diff
@@ -12279,64 +12251,14 @@ TR::Node *constrainBndChkWithSpineChk(OMR::ValuePropagation *vp, TR::Node *node)
       return node;
       }
 
-
    vp->addBlockConstraint(indexNode, constraint);
 
-   // -------------------------------------------------------------------------
-   // We can make an assertion about the minimum array size.  It must be at
-   // least as large as the low end of the index range + 1.
-   // -------------------------------------------------------------------------
-
-   if (!isSizeContigArrayLength)
-      {
-      // Contiguous arraylengths with a lower bound of zero imply the array may be
-      // discontiguous and hence the lower bound must remain at zero.
-      //
-      low = constraint->getLowInt();
-      if (!isSizeContigArrayLength || (low > 0))
-         low++;
-
-      high = static_cast<int32_t>(TR::getMaxSigned<TR::Int32>());
-      if (elementSize > 0)
-         {
-         high = high/elementSize;
-         }
-
-      constraint =  TR::VPIntRange::create(vp, low, high);
-
-      if (sizeConstraint)
-         {
-         constraint =  TR::VPIntRange::create(vp, low, high);
-         constraint = sizeConstraint->intersect(constraint, vp);
-         }
-
-      vp->addBlockConstraint(sizeNode, constraint);
-      }
-
-#if 0
-//-----------------------------------------------
-   // We can make an assertion about the minimum array size. It must be at least
-   // as large as the low end of the index range + 1.
-   //
-   low = constraint->getLowInt()+1;
-
-   high = TR::getMaxSigned<TR::Int32>();
-   if (elementSize > 0)
-      high = high/elementSize;
-   constraint =  TR::VPIntRange::create(vp, low, high);
-   if (size)
-     constraint = size->intersect(constraint, vp);
-
-   vp->addBlockConstraint(sizeNode, constraint);
-//----------------------------------------------------------
-#endif
-
    // Propagate the array size down to the underlying array object
    //
    if (sizeNode->getOpCode().isArrayLength())
       {
       TR::Node *objectRef = sizeNode->getFirstChild();
-      vp->addBlockConstraint(objectRef, TR::VPArrayInfo::create(vp, low, high, 0));
+      vp->addBlockConstraint(objectRef, TR::VPArrayInfo::create(vp, low, high + 1, 0));
```

Comment:
Could this block constraint take advantage of any constraints on the `indexNode` to further narrow the range of the array size for `objectRef`?  I notice that code in [`constrainArrayCopyBndChk`](https://github.com/eclipse-omr/omr/blob/7dbf371f9571ad8bc6326f3a78a386772e0fcccd/compiler/optimizer/VPHandlers.cpp#L12353-L12360) does that.

Reply:
I added a new commit to do this